doc/source/user_guide/io.rst

```diff
@@ -5906,6 +5907,307 @@ respective functions from ``pandas-gbq``.
+
+   Since any R object can be saved in these types, this method will only return
+   data.frame objects or objects coercible to data.frames including matrices,
+   tibbles, and data.tables.
+
+For more information of R serialization data types, see docs on `rds`_
+and `rda`_ data formats.
+
+.. _rds: https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/readRDS
+
+.. _rda: https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/save
+
+For example, consider the following generated data.frames in R using environment
+data samples from US EPA, UK BGCI, and NOAA pubilc data:
+
+.. code-block:: r
+
+   ghg_df <- data.frame(
+     gas = c("Carbon dioxide", "Methane", "Nitrous oxide",
+             "Fluorinated gases", "Total"),
+     year = c(2018, 2018, 2018, 2018, 2018),
+     emissions = c(5424.88150213288, 634.457127078267, 434.528555376666,
+                   182.782432461777, 6676.64961704959),
+     row.names = c(141:145),
+     stringsAsFactors = FALSE
+   )
+
+   saveRDS(ghg_df, file="ghg_df.rds")
+
+   plants_df <- data.frame(
+     plant_group = c("Pteridophytes", "Pteridophytes", "Pteridophytes",
+                     "Pteridophytes", "Pteridophytes"),
+     status = c("Data Deficient", "Extinct", "Not Threatened",
+                "Possibly Threatened", "Threatened"),
+     count = c(398, 65, 1294, 408, 1275),
+     row.names = c(16:20),
+     stringsAsFactors = FALSE
+   )
+
+   saveRDS(plants_df, file="plants_df.rds")
+
+   sea_ice_df_new <- data.frame(
+     year = c(2016, 2017, 2018, 2019, 2020),
+     mo = c(12, 12, 12, 12, 12),
+     data.type = c("Goddard", "Goddard", "Goddard", "Goddard", "NRTSI-G"),
+     region = c("S", "S", "S", "S", "S"),
+     extent = c(8.28, 9.48, 9.19, 9.41, 10.44),
+     area = c(5.51, 6.23, 5.59, 6.59, 6.5),
+     row.names = c(1012:1016),
+     stringsAsFactors = FALSE
+   )
+
+   saveRDS(sea_ice_df, file="sea_ice_df.rds")
+
+   save(ghg_df, plants_df, sea_ice_df, file="env_data_dfs.rda")
+
+With ``read_rdata``, you can read these above .rds or .rda files:
+
+.. ipython:: python
+   :suppress:
+
+   rel_path = os.path.join("..", "pandas", "tests", "io", "data", "rdata")
+   file_path = os.path.abspath(rel_path)
+
+.. ipython:: python
+
+   rds_file = os.path.join(file_path, "ghg_df.rds")
+   ghg_df = pd.read_rdata(rds_file)["r_dataframe"].tail()
+   ghg_df
+
+   rda_file = os.path.join(file_path, "env_data_dfs.rda")
+   env_dfs = pd.read_rdata(rda_file)
+   {k: df.tail() for k, df in env_dfs.items()}
+
+To ignore the rownames of data.frame, use option ``rownames=False``:
+
+.. ipython:: python
+
+   rds_file = os.path.join(file_path, "plants_df.rds")
+   plants_df = pd.read_rdata(rds_file, rownames=False)["r_dataframe"].tail()
+   plants_df
+
+
+To select specific objects in .rda, pass a list of names into ``select_frames``:
+
+.. ipython:: python
+
+   rda_file = os.path.join(file_path, "env_data_dfs.rda")
+   env_dfs = pd.read_rdata(rda_file, select_frames=["sea_ice_df"])
```

Comment:
Is this a good pattern?  Should it default to 

(a) reading a single frame if there is one
(b) raising and telling the user that a specific frame name is required?

or

(a) reading a dingle frame if available
(b) reading the "first" frame where first is based on either the order in the rdata file or alphabetically

Think about how this compares to `read_excel` which supports reading multiple frames from the same file?

In fact, should there be a class-based interface that would allow multiple reads from the same file, similar to `ExcelReader`?

Reply:
Originally, I had output generate either a single DataFrame for `.rds` types or dict of DataFrames for `.RData`/`.rda` types. So the file format type decided the structure. Earlier, jreback was confused by this, so now I simplified to always return a dict of DataFrames for _any_ R data file type. Because rds types carry no named objects unlike rda types, the default `r_dataframe` is used as key for single-item dict. But rda object names will map into dict keys.

For vast majority, especially in R packages, only a single named data frame is ever stored in .Rdata or .rda files. Given rdata IO may not be as popular as the Excel IO module, for class-based reader interface, maybe we should gauge this need by users for a potential future PR?

Reply:
I think that the `read_...` method should be simple and return a `DataFrame`.  Users who need to interact with complicated R data files should be directed to use the lower-level class-based reader.  This is the model that is used in `StataReader` and IIRC in `ExcelFile`.

Reply:
Hmmm...per IO docs under [specifying sheets](https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html#specifying-sheets), if you pass a list of sheets or `None`, `read_excel` returns a dict of DataFrames without specifying `ExcelFile`. This is the model used here for `read_rdata`. Also, [`read_html`](https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html#io-read-html) only returns a list of DataFrames. I thought StataReader and SASReader were more for reading files incrementally with chunks and any meta data.

Reply:
yeah for `read_hdf` we do this by forcing the user to have a key. are these always an ordered list? or a keyed list?